src/stock-on-hand-mobile/stock-on-hand-app.jsx

```diff
@@ -13,26 +13,123 @@
+                    programIds.push(permission.programId);
+                }
+            });
+        } else {
+            if (!userHomeFacility) {
+                return [];
+            }
 
-    useEffect(() => menu.style.display = '', [menu]);
+            permissions.forEach((permission) => {
+                if (userHomeFacility.id === permission.facilityId) {
+                    programIds.push(permission.programId);
+                }
+            });
+        }
+
+        const result = [];
+        programs.forEach((program) => {
+            if (programIds.indexOf(program.id) !== -1) {
+                result.push(program);
+            }
+        });
+
+        return result;
+    }
+
+    const getFacilityById = (facilities, id) => {
+        return facilities.filter(function(facility) {
+            return facility.id === id;
+        })[0];
+    }
+
+    const getSupervisedFacilities = (programId, permissions, facilities) => {
+        const facilityIds = [];
+        permissions.forEach(function(permission) {
+            if (programId === permission.programId) {
+                facilityIds.push(permission.facilityId);
+            }
+        });
+
+        const result = [];
+        facilities.forEach(function(facility) {
+            if (facilityIds.indexOf(facility.id) !== -1) {
+                result.push(facility);
+            }
+        });
+        return result;
+    }
+
+    const getSupervisedFacilitiesForAllPrograms = (programs, permissions, facilities) => {
+        const result = {};
+        programs.forEach((program) => {
+            result[program.id] = getSupervisedFacilities(program.id, permissions, facilities);
+        });
+
+        return result;
+    }
+
+    const dipatchData = (actions) => {
+        actions.forEach((action) => {
+            dispatch(action);
+        })
+    }
+
+    useEffect(() => {
+        facilityFactory.getUserHomeFacility().then((facility) =>  {
+            const userId = authorizationService.getUser().user_id;
+            asynchronousService.all([
+                facilityService.getAllMinimal(),
+                programService.getUserPrograms(userId),
+                permissionService.load(userId),
+                currentUserService.getUserInfo()
+            ])
+                .then(function(responses) {
+                    const [facilities, programs, permissions, currentUserDetails] = responses;
+
+                    const homeFacility = getFacilityById(facilities, currentUserDetails.homeFacilityId);
+                    const supervisedPrograms = getUserPrograms(true, homeFacility, permissions, programs);
+                    const supervisedFacilities = getSupervisedFacilitiesForAllPrograms(programs, permissions, facilities);
+
+                    dipatchData([setUserHomeFacilityStockOnHand(facility), setSupervisedProgramsStockOnHand(supervisedPrograms), setSupervisedFacilitiesStockOnHand(supervisedFacilities)]);
```

Comment:
Can we write this line in such way (to shorten the length of this line - I think this line is quite too long) ?
```
dipatchData([
    setUserHomeFacilityStockOnHand(facility), 
    setSupervisedProgramsStockOnHand(supervisedPrograms),
    setSupervisedFacilitiesStockOnHand(supervisedFacilities)
]);
```